src/gui/editors/PianoRoll.cpp

```diff
@@ -121,11 +121,11 @@ QPixmap * PianoRoll::s_toolOpen = NULL;
 
 TextFloat * PianoRoll::s_textFloat = NULL;
 
-static QString s_noteStrings[12] = { "C", "C#", "D", "D#", "E", "F", "F#", "G", "G#", "A", "A#", "B"};
+static QString s_noteStrings[12] = {"C", "C#", "D", "D#", "E", "F", "F#", "G", "G#", "A", "A#", "B"};
 
-static QString getNoteString( int key )
+static QString getNoteString(int key)
 {
-	return s_noteStrings[key % 12] + QString::number( static_cast<int>( key / KeysPerOctave ) );
+	return s_noteStrings[key % 12] + QString::number(static_cast<int>(FirstOctave + key / KeysPerOctave));
```

Comment:
Because the first octave starts at C-1 and the enum of octaves starts from 0. This code previously implicitly expected that the C0 octave has enum ID 0 so without the FirstOctave offset it would incorrectly start the label sequence from C0.

I initially tried to make it so that the new "minus one" octave will have index -1, but that broke a lot of other code, which says stuff like `default_key = Octave_4 * KeysPerOctave + Key_A` -- this would not produce the correct MIDI key number if the octave enum started from -1.

Reply:
Gotcha! Thanks for clarification.